File: app/src/main/java/org/amoustakos/boilerplate/examples/dao/ExampleDao.java
package org.amoustakos.boilerplate.examples.dao;

import org.amoustakos.boilerplate.examples.models.ExampleModel;
import org.amoustakos.boilerplate.injection.annotations.realm.DefaultRealm;
import org.amoustakos.boilerplate.io.db.dao.base.BaseDao;

import javax.inject.Inject;

import io.realm.Realm;
import io.realm.RealmResults;

import static org.amoustakos.boilerplate.examples.models.ExampleModel.COL_ID;
import static org.amoustakos.boilerplate.examples.models.ExampleModel.COL_NAME;


/**
 * TODO
 */
public class ExampleDao extends BaseDao<ExampleModel> {




	/*
	 * Constructors
	 */
	@Inject
	public ExampleDao(@DefaultRealm Realm realm) {
		super(realm, ExampleModel.class);
	}



	/*
	 * CRUD
	 */

	public ExampleModel getByID(String id) {
		return getOneByColumn(COL_ID, id);
	}


	public RealmResults<ExampleModel> getByName(String name) {
		return getByColumn(COL_NAME, name);
	}





}

File: app/src/main/java/org/amoustakos/boilerplate/examples/io/local/dao/ExampleDao.kt
package org.amoustakos.boilerplate.examples.io.local.dao

import io.reactivex.Observable
import io.realm.Realm
import io.realm.RealmConfiguration
import io.realm.RealmResults
import org.amoustakos.boilerplate.examples.io.local.models.ExampleModel
import org.amoustakos.boilerplate.examples.io.local.models.ExampleModel.Column.ID
import org.amoustakos.boilerplate.examples.io.local.models.ExampleModel.Column.NAME
import org.amoustakos.boilerplate.di.annotations.realm.DefaultRealmConfig
import org.amoustakos.boilerplate.io.realm.dao.base.BaseDao
import org.amoustakos.utils.android.rx.RxUtil


class ExampleDao(realm: Realm):
		BaseDao<ExampleModel>(realm, ExampleModel::class.java) {


    fun getByID(id: String): ExampleModel? = getOneByColumn(ID, id)

    fun getByName(name: String): RealmResults<ExampleModel> =
            getByColumn(NAME, name)



    companion object {

        fun getInstance(@DefaultRealmConfig config: RealmConfiguration) =
                ExampleDao(Realm.getInstance(config))

        fun getObservableInstance(
                @DefaultRealmConfig config: RealmConfiguration): Observable<ExampleDao> =
		        Observable.just {}
                        .compose(RxUtil.applyRealmObservableScheduler())
				        .map { ExampleDao(Realm.getInstance(config)) }


    }
}
